### ISSUE TITLE
bug: nvim-cmp Completion Popup Overlaps with LSP Hover Popup in LazyVim

### ISSUE THREAD

**Issue by xinghe98** [LazyVim/LazyVim#4736]
### Did you check docs and existing issues?

- [x] I have read all the LazyVim docs
- [x] I have updated the plugin to the latest version before submitting this issue
- [x] I have searched the existing issues of LazyVim
- [x] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

0.10.2

### Operating system/version

windows11/wsl2-arch

### Describe the bug

 When typing in LazyVim, the completion popup from nvim-cmp and the LSP hover documentation popup often overlap, especially when certain characters are typed (like <). The completion window (cmp) should take priority over the hover window, but currently, both are shown simultaneously, causing visual clutter and confusion during coding.![Image](https://github.com/user-attachments/assets/b1579983-74c0-4ab7-b3e2-9691730d582b)

### Steps To Reproduce

1. Install and use LazyVim with the default configuration.
2. Open a TypeScript/JavaScript/React file.
3. Start typing a JSX tag like <App.
4. Notice that both the LSP hover popup and the cmp completion popup appear simultaneously, resulting in overlapping windows.

### Expected Behavior

The cmp completion window should take precedence over the LSP hover window when a completion is in progress. Ideally, the hover window should either:

- Be hidden when the cmp window is active.
- Be shown only after a completion selection is made (if the user wants to see hover details for the selected item).

Possibly add an option to delay or suppress the hover popup when the cmp window is active.

### Repro

```lua
vim.env.LAZY_STDPATH = ".repro"
load(vim.fn.system("curl -s https://raw.githubusercontent.com/folke/lazy.nvim/main/bootstrap.lua"))()

require("lazy.minit").repro({
  spec = {
    { "LazyVim/LazyVim", import = "lazyvim.plugins" },
    -- add any other plugins here
  },
})
```

**Comment by dpetka2001** [comment]
The problem you observe here has to do with the way `noice.nvim` is showing the LSP signature while typing. There are issues about it on `noice.nvim` repo, so you should also comment on one of the issues, because it won't get fixed here. You can temporarily disable `opts.lsp.signature.auto_open.enabled = false` and manually call it when you want to check it.

**Comment by xinghe98** [comment]
> The problem you observe here has to do with the way `noice.nvim` is showing the LSP signature while typing. There are issues about it on `noice.nvim` repo, so you should also comment on one of the issues, because it won't get fixed here. You can temporarily disable `opts.lsp.signature.auto_open.enabled = false` and manually call it when you want to check it.

I noticed the issue you mentioned regarding the way noice.nvim shows LSP signatures while typing. I was wondering if you happen to have a link to one of the related issues in the noice.nvim repository. I’d love to contribute or follow along with the discussion.

If there isn’t one available, I’m happy to create a new issue myself.

Thanks so much for your help!

**Comment by dpetka2001** [comment]
[This one](https://github.com/folke/noice.nvim/issues/930) seems relevant although I'm not sure if it's exactly what you're experiencing. But it might have to do with the same root cause. You might want to follow up on that issue.

**Comment by xinghe98** [comment]
thanks!
